pkg/splunk/enterprise/clustermanager.go

```diff
@@ -434,3 +451,160 @@ func VerifyCMisMultisite(ctx context.Context, cr *enterpriseApi.ClusterManager,
+		scopedLog.Error(err, "StatefulSet types not found in namespace", "namsespace", cr.GetNamespace())
+		return "", err
+	}
+	labelSelector, err := metav1.LabelSelectorAsSelector(statefulSet.Spec.Selector)
+	if err != nil {
+		eventPublisher.Warning(ctx, "getClusterManagerCurrentImage", fmt.Sprintf("Could not get labels. Reason %v", err))
+		scopedLog.Error(err, "Unable to get labels")
+		return "", err
+	}
+
+	// get a list of all pods in the namespace with matching labels as the statefulset
+	statefulsetPods := &corev1.PodList{}
+	opts := []rclient.ListOption{
+		rclient.InNamespace(cr.GetNamespace()),
+		rclient.MatchingLabelsSelector{Selector: labelSelector},
+	}
+
+	err = c.List(ctx, statefulsetPods, opts...)
+	if err != nil {
+		eventPublisher.Warning(ctx, "getClusterManagerCurrentImage", fmt.Sprintf("Could not get Pod list. Reason %v", err))
+		scopedLog.Error(err, "Pods types not found in namespace", "namsespace", cr.GetNamespace())
+		return "", err
+	}
+
+	// find the container with the phrase 'splunk' in it
+	for _, v := range statefulsetPods.Items {
+		for _, container := range v.Status.ContainerStatuses {
+			if strings.Contains(container.Name, "splunk") {
+				image := container.Image
+				return image, nil
+			}
+
+		}
+	}
+
+	return "", nil
+}
+
+// changeClusterManagerAnnotations updates the checkUpdateImage field of the CLuster Manager Annotations to trigger the reconcile loop
+// on update, and returns error if something is wrong
+func changeClusterManagerAnnotations(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.LicenseManager) error {
+
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("changeClusterManagerAnnotations").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	eventPublisher, _ := newK8EventPublisher(c, cr)
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      cr.Spec.ClusterManagerRef.Name,
+	}
+	clusterManagerInstance := &enterpriseApi.ClusterManager{}
+	err := c.Get(ctx, namespacedName, clusterManagerInstance)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return nil
+	}
+
+	image, _ := getLicenseManagerCurrentImage(ctx, c, cr)
+
+	// fetch and check the annotation fields of the ClusterManager
+	annotations := clusterManagerInstance.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+	if _, ok := annotations["checkUpdateImage"]; ok {
+		if annotations["checkUpdateImage"] == image {
+			return nil
+		}
+	}
+
+	// create/update the checkUpdateImage annotation field
+	annotations["checkUpdateImage"] = image
+
+	clusterManagerInstance.SetAnnotations(annotations)
+	err = c.Update(ctx, clusterManagerInstance)
+	if err != nil {
+		eventPublisher.Warning(ctx, "changeClusterManagerAnnotations", fmt.Sprintf("Could not update annotations. Reason %v", err))
+		scopedLog.Error(err, "ClusterManager types update after changing annotations failed with", "error", err)
+		return err
+	}
```

Comment:
Maybe this whole piece of code can be a common code since it is same in `changeMonitoringConsoleAnnotations` as well? So all these lines can be a part of a generic API which can be called from both `changeMonitoringConsoleAnnotations` and  `changeClusterManagerAnnotations` ?